### DISCUSSION TITLE
Notice Subtype 14, BT-14-Lot,BT-15-Lot and BT-615-Lot

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#408] [Q&A] [answered]
Hi Lab
one question :
in Notice 14,  BT-14-Lot is not mandatory, while BT-15-Lot and BT-615-Lot are.
as result , choosing in Front-end just the mandatory fields ,  I get the following  xml section, that of course is not xsd compliant : 
<cac:CallForTendersDocumentReference>
				<cbc:ID>MARIO_PIO</cbc:ID>
				<cac:Attachment>
					<cac:ExternalReference>
						<cbc:URI>http://sdfsds.sfdf</cbc:URI>
					</cac:ExternalReference>
				</cac:Attachment>
				<cac:Attachment>
					<cac:ExternalReference>
						<cbc:URI>http://sdfsds.sfdf</cbc:URI>
					</cac:ExternalReference>
				</cac:Attachment>
			</cac:CallForTendersDocumentReference>
As   Attachment is not mandatory, (**<xsd:element ref="cac:Attachment" minOccurs="0" maxOccurs="1"/>**) What about set all three fields not mandatory ? 
Thanks and Regards
Renato

**Comment by YvesJo** [comment, marked as answer]
Hi Renato,
BT-15 and BT-615 are exclusive, as soon as one is selected the other one is not allowed (a document may not be simultaneously restricted and non-restricted).
The XSD is used for lexical validation while the schematrons are used for validation (including existence requirements). Requirements expressed in the schematron are stronger than in the XSD and deal with context related cases.
As soon as a document is identified, it must be specified whether it is restricted or not and the corresponding URL has to be specified.
I hope this answers your question.
KR,
Yves

**Reply by rengit62** [reply]
Hi Yves
BT-15 and BT-615 are both mandatory in the form, I need to know how a java program can choose between them.
Perhaps during xml creation the program could perform a xpath search to exclude one of them : 

from fields.json : 

  "id" : "BT-615-Lot",
**************************  
    "xpathRelative" : "cac:Attachment[../cbc:DocumentType/text()='restricted-document']/cac:ExternalReference/cbc:URI",

 "id" : "BT-15-Lot",
************************* 
    "xpathRelative" : "cac:Attachment[not(../cbc:DocumentType/text()='restricted-document')]/cac:ExternalReference/cbc:URI",
   
but , if I understood well, this way was not suggested in :   #405 

  
so either I did not understand well in #405 or please suggest me how a java program can exclude one of those fields, as in the form they are both mandatory

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/ea046887-d1bb-48d1-b529-bebff9e3ff05)

Thanks and Regards !
Renato

**Reply by YvesJo** [reply]
Both fields are conditionally mandatory, they may each only exist under one specific condition, the diffusion of the procurement is restricted or it is not restricted. BT-15 and BT-615 may not coexist but when one of these may exist (conditionally forbidden rule allowing it) then is is also mandatory (conditionally mandatory rule sharing the same condition but in its negated form).
